lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -1182,6 +1182,17 @@ def modify_connection(self):
             cmd = self.modify_connection_bridge_slave()
         elif self.type == 'vlan':
             cmd = self.modify_connection_vlan()
+            # If no dns given, then we assume that IPv4/IPv6 is not used on the interface
+            # so we are going to disable the protocol, otherwise interface won't come up.
+
+            if self.ip4 is 'disabled':
+                self.execute_command(cmd)
+                cmd = self.modify_conn_disable_ipv4()
+
+            if self.ip6 is 'ignore':
```

Comment:
Does this functionality only affects VLAN ? What about other connection types and modes like ethernet and bridge etc.,?

Reply:
Again string comparison with `is`.

Reply:
Good catch @thomwiggers